### PR TITLE
Fixed several type definitions

### PR DESCRIPTION
**Pull request by franklixuefei**
**What**:
Fixed several type definitions as well as added the missing ones.

**Why**:
For proper typescript support.

**How**:
Please take a look at the commits.

**Checklist**:

* [ ] Documentation
* [ ] Tests
* [x] Ready to be merged
      
* [x] Added myself to contributors table

### CODE REVIEW

**Comment by franklixuefei** [comment]
Fixes #406

File: typings/index.d.ts [hunk 1]
```diff
@@ -132,20 +133,20 @@ export interface PropGetters<Item> {
 }
 
 export interface Actions<Item> {
-  reset: (otherStateToSet?: {}, cb?: Callback) => void
+  reset: (otherStateToSet?: StateChangeOptions<Item>, cb?: Callback) => void
```
Comment (stereobooster) [inline typings/index.d.ts]: Should it be `StateChangeOptions<Item>` or `Partial<StateChangeOptions<Item>>`?

**Comment by stereobooster** [comment]
Hi @franklixuefei thanks for contribution. I have one question otherwise looks ok

File: typings/index.d.ts (hunk 1 above)
Comment (franklixuefei) [reply typings/index.d.ts]: Good catch! Yes indeed it should be `Partial`

**Comment by franklixuefei** [comment]
@stereobooster Fixed the comment. Good catch.

**Comment by kentcdodds** [comment]
Really hoping we can have a beta 2.0.0 release in the next week or two. I'm just super busy at the moment :-/

**Comment by franklixuefei** [comment]
@kentcdodds That'll be awesome!

**Pull request merged**